### PR TITLE
feat: Partial CouchDB implementation of new storage interface

### PR DESCRIPTION
**Pull request by DRK3**
Remaining methods to be implemented in further PRs.

Signed-off-by: Derek Trider <Derek.Trider@securekey.com>

### CODE REVIEW

**Comment by DRK3** [comment]
FYI @pfeairheller - this should begin to give you an idea of how the new storage interface will work

File: test/component/newstorage/newstore.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	"errors"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/hyperledger/aries-framework-go/pkg/newstorage"
+	"github.com/stretchr/testify/require"
+)
+
+// TestAll tests common storage functionality.
+func TestAll(t *testing.T, provider newstorage.Provider) {
```
Comment (DRK3) [inline test/component/newstorage/newstore.go:20]: When I work on other storage implementations in this repo, I'll add more common tests here

**Comment by soluchok** [comment]
@DRK3 you need to set up `workflows` as well. 
See https://github.com/hyperledger/aries-framework-go-ext/blob/main/.github/workflows/storage-couchdb.yml

File: component/newstorage/couchdb/store_internal_test.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
+/*
```
Comment (soluchok) [inline component/newstorage/couchdb/store_internal_test.go:1]: The name of the file should be `store_test.go`

File: component/newstorage/couchdb/store.go [hunk 3]
```diff
@@ -0,0 +1,604 @@
...
+				// Need to get the new revision ID and try again.
+				return fmt.Errorf(failPutValueViaClient, err)
+			}
+
+			err = fmt.Errorf(failPutValueViaClient, err) // Wrap error here and return below.
+		}
+
+		// If there is no error or it's something other than an update conflict, then no more retries are needed.
+		// We return a backoff.Permanent wrapped error to prevent further retries.
+		return backoff.Permanent(err)
```
Comment (Baha-sk) [inline component/newstorage/couchdb/store.go:431]: err is `nil` here, set `nil` to make it explicit

**Review by Baha-sk — APPROVED** [review]

**Review by sandrask — APPROVED** [review]

File: component/newstorage/couchdb/store_internal_test.go (hunk 2 above)
Comment (DRK3) [reply component/newstorage/couchdb/store_internal_test.go:1]: There's already a `store_test.go` in this PR. So this PR has two test files in `newstorage/couchdb`: `store_test.go` and `store_internal_test.go`.

`store_test.go` is a separate package and it's used to run tests against exported stuff only. My goal was to increase code coverage as much as possible using only exported stuff. However, I couldn't reach 85% coverage this way. This is where `store_internal_test.go` comes in - it's in the same package as `store.go` and allows me to test error cases that require me to mock internal components.

File: component/newstorage/couchdb/store.go (hunk 3 above)
Comment (DRK3) [reply component/newstorage/couchdb/store.go]: Done

**Review by fqutishat — APPROVED** [review]

**Pull request merged**